sycl/include/sycl/ext/oneapi/matrix/matrix-unified.hpp

```diff
@@ -112,6 +112,29 @@ get_wi_data(Group sg, joint_matrix<Group, T, Use, Rows, Cols, Layout> &jm) {
 #endif // defined(__SYCL_DEVICE_ONLY__)
 }
 
+template <typename Group, typename T, use Use, size_t M, size_t N,
+          layout Layout, typename F>
+inline __SYCL_ALWAYS_INLINE void
+joint_matrix_apply(Group sg, joint_matrix<Group, T, Use, M, N, Layout> &jm,
+                   F &&lambda) {
+#if defined(__SYCL_DEVICE_ONLY__)
+#if defined(__NVPTX__)
+  for (int i = 0; i < jm.cuda_impl.wi_marray.size(); i++) {
+    lambda(jm.cuda_impl.wi_marray[i]);
+  }
+#else // NVPTX
+  auto wi_data_c = sycl::ext::intel::experimental::matrix::get_wi_data(sg, jm);
+  for (int i = 0; i < wi_data_c.length(); i++) {
+    lambda(wi_data_c[i]);
```

Comment:
wi_data's operator[] is returning a temporary object but lambda take a lvalue as input. it will lead to compfail.
the following code can work:
  auto wi_data_c = sycl::ext::intel::experimental::matrix::get_wi_data(sg, jm);
  for (int i = 0; i < wi_data_c.length(); i++) {
    auto wie = wi_data_c[i];
    lambda(wie);
  }

Reply:
Besides this, this should be added to update joint matrix values:
"wi_data_c[i] = wie;"
Also, C should be passed by reference to joint_matrix_apply

Reply:
This is the full code that runs correctly:
```c++
void joint_matrix_apply(Group sg, joint_matrix<Group, T, Use, M, N, Layout>& C, F&& lambda) {
     auto wi_data_c = get_wi_data(sg, C);
     for (int i = 0; i < wi_data_c.length(); i++) {
       T wie = wi_data_c[i];
       lambda(wie);
       wi_data_c[i] = wie;
     }
     return;
}
```

Reply:
Updated